module-4/cdk/lib/APIGatewayConstruct.ts

```diff
@@ -0,0 +1,73 @@
+import apigateway = require('@aws-cdk/aws-apigateway');
+import elbv2 = require('@aws-cdk/aws-elasticloadbalancingv2');
+import cdk = require('@aws-cdk/cdk');
+import fs = require('fs');
+import path = require('path');
+import { EcsClusterConstruct } from '../../../module-2/cdk/lib/ECSConstruct';
+
+export class APIGatewaySchemaConstruct extends cdk.Construct {
+  constructor(scope: cdk.Construct, id: string, ecsCluster: EcsClusterConstruct) {
+    super(scope, id);
+
+    const nlb = elbv2.NetworkLoadBalancer.fromNetworkLoadBalancerAttributes(this, 'NLB', {
+      loadBalancerArn: ecsCluster.ecsService.loadBalancer.loadBalancerArn,
+    });
+    const vpcLink = new apigateway.VpcLink(this, 'VPCLink', {
+      description: 'VPC Link for our  REST API',
+      name: 'MysfitsApiVpcLink',
+      targets: [
+        nlb
+      ]
+    });
+    const schema = this.generateSwaggerSpec(ecsCluster.ecsService.loadBalancer.loadBalancerDnsName, vpcLink);
+    if (schema) {
+      const jsonSchema = JSON.parse(schema);
+      const api = new apigateway.CfnRestApi(this, 'Schema', {
```

Comment:
Can we use L2 without swagger here? Or we can add support for swagger?

Reply:
Easiest option was to use the Swagger they previously had.  Adding Swagger support would be awesome.